archive/tar.go

```diff
@@ -106,13 +117,13 @@ func Apply(ctx context.Context, root string, r io.Reader) (int64, error) {
 
 	// Iterate through the files in the archive.
 	for {
+		hdr, err := tr.Next()
```

Comment:
Why move above? Is there a reason we would want to do the done check after the blocking call rather than before?

Reply:
Not intentional. I moved this when attempting to combine the tar extract loop with Windows apply, but decided against it. I'll move it back where it was.